### ISSUE TITLE
MPI error on HPC platforms

### ISSUE THREAD

**Issue by SorooshMani-NOAA** [schism-dev/RiverMeshTools#7]
When I try to use RiverMeshTools, and in specific `pyDEM` on some of our HPC platforms I get errors like:

```
>>> from pyDEM import dem
[sorooshmani-nhccolab2-00010-1-0002:16095] OPAL ERROR: Unreachable in file pmix3x_client.c at line 112
--------------------------------------------------------------------------
The application appears to have been direct launched using "srun",
but OMPI was not built with SLURM's PMI support and therefore cannot
execute. There are several options for building PMI support under
SLURM, depending upon the SLURM version you are using:

  version 16.05 or later: you can use SLURM's PMIx support. This
  requires that you configure and build SLURM --with-pmix.

  Versions earlier than 16.05: you must use either SLURM's PMI-1 or
  PMI-2 support. SLURM builds PMI-1 by default, or you can manually
  install PMI-2. You must then build Open MPI using --with-pmi pointing
  to the SLURM PMI library location.

Please configure as appropriate and try again.
--------------------------------------------------------------------------
*** An error occurred in MPI_Init_thread
*** on a NULL communicator
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)
```

This issue seems to be coming from `pylibs` library. Because if I comment the line where it says
```
from pylib import *
```

Then I can import `pyDEM` (of course I cannot use it this way!). I was not sure if I should create this ticket here or on @wzhengui's repo at https://github.com/wzhengui/pylibs

**Update**
Note that I don't have this issue on my laptop. So I think it's specific to MPI installations on the system and how it is set up. Any help in how to avoid this is appreciated.

**Comment by SorooshMani-NOAA** [comment]
Also this is from @bahramkhazaei-noaa on Hera 

```
python /river-in-mesh/river_in_mesh/scripts/extract_rivers.py -h
Abort(1090831) on node 0 (rank 0 in comm 0): Fatal error in PMPI_Init_thread: Other MPI error, error stack:
MPIR_Init_thread(176): 
MPID_Init(1430)......: 
MPIR_pmi_init(167)...: PMI2_Job_GetId returned 14
```

**Comment by SorooshMani-NOAA** [comment]
The issue seems further upstream actually. Even trying to `from mpi4py import MPI` fails on our HPC platforms. How do you install `mpi4py` on your HPC? Do you use a HPC provided library or do you compile it yourself?

**Comment by SorooshMani-NOAA** [comment]
Thanks! I'll try a few things, if I can come up with a generic approach I'll post it here. So far I've been unsuccessful in getting it to work!

**Comment by SorooshMani-NOAA** [comment]
@cuill I tried installing `mpi4py` after loading our HPC installed `openmpi` and `gnu` compiler and then called `pip install --no-cache-dir --no-binary :all: mpi4py`. Although I'm not sure if this actually had anything to do with resolving the issue.

Then when getting an interactive session, instead of calling `srun ...` I just allocated the resource using `salloc` and then `ssh`ed into the allocation. 

Basically the main issue seems to have been the `srun`. My understanding is that when we get an allocation with interactive session, we have to make sure that the session is not initiated with `srun` for things to work properly with `mpi4py`! 

I'll update this comment as I test more. Note that I haven't yet tested running MPI jobs with this installation, I was just trying to interactively use `pyDEM` stuff. For now I close this since the main issue is kind of addressed!
